src/metabase/driver/postgres.clj

```diff
@@ -758,3 +760,29 @@
   [driver prepared-statement i t]
   (let [local-time (t/local-time (t/with-offset-same-instant t (t/zone-offset 0)))]
     (sql-jdbc.execute/set-parameter driver prepared-statement i local-time)))
+
+(defn- load-from-csv-sql [schema-name table-name column-names file-name]
+  (str "COPY " schema-name "." table-name "(" (str/join "," column-names) ") FROM '" file-name
+       "' WITH (FORMAT CSV, HEADER TRUE, ENCODING 'UTF8', QUOTE '\"', ESCAPE '\\')"))
+
+(def csv->database-type
+  {::csv/varchar_255 "VARCHAR(255)"
+   ::csv/text        "TEXT"
+   ::csv/int         "INTEGER"
+   ::csv/float       "FLOAT"
+   ::csv/boolean     "BOOLEAN"})
+
+(defmethod driver/load-from-csv :postgres
+  [driver db-id schema-name table-name file-name]
+  (let [csv-schema (csv/detect-schema file-name)
+        csv-schema (for [[k v] csv-schema]
+                     {:database-name k
+                      :csv-type      v
+                      :database-type (csv->database-type v)})
+        cols       (map :database-name csv-schema)
+        table-name (str table-name "_" (t/format "yyyyMMddHHmmss" (t/local-date-time)))]
+    (driver/create-table driver db-id schema-name table-name csv-schema)
+    (let [sql (load-from-csv-sql schema-name table-name cols file-name)
+          upload-result (qp.writeback/execute-write-sql! db-id sql)]
+      (when (pos? (:rows-affected upload-result))
+        table-name))))
```

Comment:
it can be in a follow-up PR, but we need to get rid of the new table if the CSV loading fails